### DISCUSSION TITLE
Data output

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#25332] [Q&A Modules: General]
Hello,

I have a question concerning the number of qp per element; when considering quad4 for elements and looking at the output through paraview I can see that Von Mises stress is defined as cell data (one value per element) while displacement which is a variable computed at nodes is a point data (with 4 values per element), I would like to know what these 4 values represent. 

Regards

**Comment by GiudGiud** [comment]
Hello

When  displacing a mesh, you move the nodes. That s what the displacement represent.
The number of qps depends on the quadrature used. By default we use a Gauss one such that it can integrate order_of_variables * 2 + 1
https://mooseframework.inl.gov/syntax/Executioner/Quadrature/index.html

Guillaume

**Comment by ghost** [comment]
We have added a material property to our material model
```
_cp(declareProperty<RealVectorValue>("cp"))
```
This cp is defined at qp.
When trying to print out this material property we get 4 different values for one quad4 element.
```
std::cout << "cp  :   "<< _cp[_qp](0)<< " " << _cp[_qp](1) << " " <<_cp[_qp](2)<< std::endl ;
```
When opening the exodus file with paraview we find one single value per element (defined as a cell data). 
So we do not understand what this one value per element represents.

**Reply by GiudGiud** [reply]
It s the weighted average iirc

**Reply by ghost** [reply]
So this is just a paraview (or peacock) averaging?

**Reply by GiudGiud** [reply]
no it s MOOSE outputting the value on a constant monomial variable

you can get Paraview to smooth it out by converting it to a nodal value btw

**Reply by ghost** [reply]
So you confirm that for example with a quad4 element   4 integration points (by default)  are considered in the computation but only one value is output ?

**Reply by GiudGiud** [reply]
yes. 
4 qp is the default for your variable order. If you bump that up you ll see more

**Reply by ghost** [reply]
ok. thank you
